nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/StandardSchemaIdentifier.java

```diff
@@ -25,15 +25,18 @@
     private final Optional<String> name;
     private final OptionalLong identifier;
     private final OptionalInt version;
+    private final OptionalLong schemaVersionId;
     private final Optional<String> branch;
 
-    StandardSchemaIdentifier(final String name, final Long identifier, final Integer version, final String branch) {
+    StandardSchemaIdentifier(final String name, final Long identifier, final Integer version,
+            final Long schemaVersionId, final String branch) {
         this.name = Optional.ofNullable(name);
-        this.identifier = identifier == null ? OptionalLong.empty() : OptionalLong.of(identifier);;
-        this.version = version == null ? OptionalInt.empty() : OptionalInt.of(version);;
+        this.identifier = identifier == null ? OptionalLong.empty() : OptionalLong.of(identifier);
+        this.version = version == null ? OptionalInt.empty() : OptionalInt.of(version);
+        this.schemaVersionId = schemaVersionId == null ? OptionalLong.empty() : OptionalLong.of(schemaVersionId);
         this.branch = Optional.ofNullable(branch);
 
-        if (this.name == null && this.identifier == null) {
+        if ((this.name == null && this.identifier == null) || this.schemaVersionId == null) {
```

Comment:
This logic seems to be off a bit. It indicates that `this.schemaVersionId` cannot be null. And either `this.name` or `this.identifier` cannot be null. But none of them can be null because they are set to Optional objects above...

Reply:
Since we have a constant for an EMPTY schema:
`SchemaIdentifier EMPTY = new StandardSchemaIdentifier(null, null, null, null, null);`
I think it makes more sense to get rid of this check all together since it wasn't actually doing anything before.

Reply:
Pushed a new commit